### PR TITLE
feat(notifications): Add the ability to disable workflow notifications

### PR DESCRIPTION
**Pull request by tkaemming**
This is a rebased version of GH-4681.

### CODE REVIEW

**Comment by tkaemming** [comment]
Didn't add anybody to review this yet since I'm guessing between the rebase and 10 months of waiting something has probably broken

File: CHANGES [hunk 1]
```diff
@@ -342,6 +342,7 @@ Version 8.12
 - Fixed usage (and propagation) of ``Group.first_release``.
 - The + and - datetime search helpers now work with ranges (e.g. ``<=``).
 - Added the ability to download artifacts from releases.
+- Added the ability for users to disable workflow notifications on a per-project basis.
```
Comment (tkaemming) [inline CHANGES]: Guess I should move this, huh

File: src/sentry/models/useroption.py [hunk 2]
```diff
@@ -21,6 +21,7 @@ class UserOptionValue(object):
     # 'workflow:notifications'
     all_conversations = '0'
     participating_only = '1'
+    no_conversations = '2'
```
Comment (tkaemming) [inline src/sentry/models/useroption.py]: These ID spaces are disjoint, overlapping here doesn't matter

**Review by tkaemming — COMMENTED** [review]
Did a cursory pass over this and things seem to be working in the UI. Tests should validate that the more complex logic is correct as well.

**Comment by tkaemming** [comment]
Also, the notification settings page looks kinda bad now and could probably use some tidying up

**Comment by billyvg** [comment]
From looking at percy, there seems to be a separator missing from the "Workflow" header

File: src/sentry/templates/sentry/account/notifications.html [hunk 3]
```diff
@@ -60,23 +60,22 @@
...
+            Workflow notifications are separate from alerts and are generated
+            for issue updates, such as changes in issue assignment, changes to
+            resolution status (including regressions), and comments.
+          {% endblocktrans %}
+        </p>
         <p>
           {% blocktrans %}
             When workflow notifications are enabled for a project, you'll
             receive an email when your teammates perform any of these actions.
+            You'll be automatically be added as a participant on an issue by
```
Comment (macqueen) [inline src/sentry/templates/sentry/account/notifications.html]: `You'll be automatically be added`

Comment (tkaemming) [reply src/sentry/templates/sentry/account/notifications.html]: 😬

Comment (tkaemming) [reply src/sentry/templates/sentry/account/notifications.html]: 0465cc55a5cb4d88ba0bbb22b9dc2e436582a80e

**Comment by tkaemming** [comment]
> From looking at percy, there seems to be a separator missing from the "Workflow" header

7db73f7d837a153a6eb379099d0d69e3903829c2

**Review by macqueen — APPROVED** [review]

**Pull request merged**